File: src/main/java/com/mycompany/moviecrosssearch/SaveDBTask.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.moviecrosssearch;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author flatba
 */
public class SaveDBTask {

    ContentInformationModel cim;

    SaveDBTask(ContentInformationModel model) {
            cim = new ContentInformationModel();
            cim = model;
    }

    private static Connection conn;
    private static Statement stmt;
//    private static ResultSet rs;

//	private void saveDBSqlite() { // ファイル単体のデバッグ中なのでコメントアウト
    public static void main(String[] args) {

	String outputDBDir = "jdbc:sqlite:" + "/Users/flatba/dev/output/testdatabase/sample.db";

	try {
            Class.forName("org.sqlite.JDBC");
            conn = DriverManager.getConnection(outputDBDir);
            stmt = conn.createStatement();

            //テーブル作成
            stmt.executeUpdate(
                            "create table test1"
                            + "( "
                            + "title string, "
                            + "originalTitle string, "
                            + "rereaseYear integer, "
                            + "genre string, "
                            + "runningTime string, "
                            + "director string, "
                            + "summary text"
                            + ")"
            );

            //値を入力する
            stmt.execute( "insert into test1 values ( 'テストタイトル', '原題', 2017, 'ジャンル', '上映時間', '監督', 'これはあらすじです。' )" );

            //結果を表示する
//            rs = stmt.executeQuery("select * from test1");
//            while(rs.next()) {
//                System.out.println(rs.getString("title"));
//                System.out.println(rs.getInt("originalTitle"));
//                System.out.println(rs.getInt("rereaseYear"));
//                System.out.println(rs.getInt("genre"));
//                System.out.println(rs.getInt("runningTime"));
//                System.out.println(rs.getInt("director"));
//                System.out.println(rs.getInt("summary"));
//            }

            } catch (ClassNotFoundException e) {
                // TODO 自動生成された catch ブロック
                e.printStackTrace();
            } catch (SQLException e) {
                // TODO 自動生成された catch ブロック
                e.printStackTrace();
            } finally {
                if(conn != null) {
                    try {
                        //接続を閉じる
                        conn.close();
                    } catch (SQLException e) {
                        // TODO 自動生成された catch ブロック
                        e.printStackTrace();
                    }
                }
            }
    }

}

File: SaveDBTask.rb
# coding: utf-8
require 'sqlite3'

#
# 保存処理
#
class SaveDBTask

  #
  # DB生成
  # CREATE TABLE IF NOT EXISTS をすると作成済みのテーブルを作ろうとするエラーを防げます。
  #
  def initialize(site_name)
    output_directory = 'output/' + site_name + '.db'
    @db = SQLite3::Database.new(output_directory)
    @db.execute(
      'CREATE TABLE IF NOT EXISTS contents_master (
      thumbnail varchar(200),
      title varchar(200),
      original_title varchar(200),
      release_year varchar(200),
      genres varchar(200),
      running_time varchar(200),
      director varchar(200),
      summary varchar(200)
      );'
    )
  end

  def saveDBTask(contents)
    @db.execute(
      "insert into contents_master (
      thumbnail, title, original_title, release_year, genres, running_time, director, summary
      ) values(
      '#{contents.thumbnail}',
      '#{contents.title}',
      '#{contents.original_title}',
      '#{contents.release_year}',
      '#{contents.genres}',
      '#{contents.running_time}',
      '#{contents.director}',
      '#{contents.summary}'
    )")
  end

  def closeDBTask
    @db.close
  end
end